### PR TITLE
Geometrical Shoot Angle Equation

### PR DESCRIPTION
**Pull request by bengrif**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java [hunk 1]
```diff
@@ -24,6 +24,13 @@
 import java.util.function.DoubleSupplier;
 
 public class ArmSubsystem extends SubsystemBase {
+  private static final double ROBOT_TO_SHOOTER_PIVOT = .2;
+  private static final double SPEAKER_X_OFFSET = .23;
+  private static final double SPEAKER_Z_HEIGHT = 2.032;
+  private static final double SHOOTER_ANGLE = 56.0;
+  private static final double SHOOTER_BASE_HEIGHT = .42;
+  private static final double ARM_LENGTH = .425;
```
Comment (Kellenn) [inline src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java:32]: Let's move these to ``ArmConstants``

File: src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java [hunk 2]
```diff
@@ -286,6 +293,40 @@ private double getPositionFromDistance(double distance) {
...
+    double angleC = 180 - SHOOTER_ANGLE - Math.toDegrees(angleB);
+
+    /*
+     * Calculate the angle of the pivot point up to the speaker, add angleC to compute the arm angle.
+     */
+    double angle = Math.toDegrees(Math.atan(height / distance)) + angleC;
+
+    angle = 180 - angle;
+
+    return angle;
```
Comment (Kellenn) [inline src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java:327]: To stay in line with our other code, let's return a ``Rotation2d`` here.

File: src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java [hunk 3]
```diff
@@ -286,6 +293,40 @@ private double getPositionFromDistance(double distance) {
...
   }
 
+  /*
+   * Calculates the angle of the speaker to the arm pivot point plus the angle of the shooter.
+   *
+   */
+  public static double calculateAngleFromDistance(double distance) {
+
+    distance = distance - ROBOT_TO_SHOOTER_PIVOT - SPEAKER_X_OFFSET;
+    final double height = SPEAKER_Z_HEIGHT - SHOOTER_BASE_HEIGHT;
```
Comment (Kellenn) [inline src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java:303]: Why final?

File: src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java [hunk 4]
```diff
@@ -286,6 +293,40 @@ private double getPositionFromDistance(double distance) {
     }
   }
 
+  /*
+   * Calculates the angle of the speaker to the arm pivot point plus the angle of the shooter.
+   *
+   */
+  public static double calculateAngleFromDistance(double distance) {
```
Comment (Kellenn) [inline src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java:300]: This doesn't need to be made public or declared as static. We also never utilize it.

**Review by Kellenn — CHANGES_REQUESTED** [review]
A little bit of a pedant here :D

File: src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java (hunk 4 above)
Comment (bengrif) [reply src/main/java/frc/robot/subsystems/manipulator/ArmSubsystem.java:300]: It was static to make it easier to use in a test routine.   Agreed it doesn't need to be static, but also doesn't need to be non-static :) .

**Review by Kellenn — APPROVED** [review]

**Comment by Kellenn** [comment]
Obsoleted by #30

**Pull request closed without merging**